ocaml/tests/test_cluster_host.ml

```diff
@@ -122,11 +122,91 @@ let test_destroy_forbidden_when_sr_attached () =
     Api_errors.(Server_error (cluster_stack_in_use, [ cluster_stack ]))
     (fun () -> Xapi_cluster_host.destroy ~__context ~self:cluster_host)
 
+type declare_dead_args = {
+  dead_members: Cluster_interface.address list;
+  dbg: string
+} [@@deriving rpcty]
+
+let test_clusterd_rpc ~__context call =
+  match call.Rpc.name, call.Rpc.params with
+  | "declare-dead", [args] ->
+    let args = Rpcmarshal.unmarshal declare_dead_args.Rpc.Types.ty args |> Rresult.R.get_ok in
+    let all = Db.Cluster_host.get_all ~__context in
+    let ndead = List.length args.dead_members in
+    let nall = List.length all in
+    Printf.printf "dead_members: %d, all: %d\n" ndead nall;
+    if ndead = nall - 1 then
+      Rpc.{success = true; contents = Rpc.rpc_of_unit () }
+    else
+      let err = Cluster_interface.InternalError "Remaining hosts are not all alive" in
+      (* in the test we must declare N-1 as dead before it succeeds *)
+      Rpc.failure (Rpcmarshal.marshal Cluster_interface.error.Rpc.Types.ty err)
+  | name, params ->
+    failwith (Printf.sprintf "Unexpected RPC: %s(%s)" name (String.concat " " (List.map Rpc.to_string params)))
+
+let test_rpc ~__context call =
+  match call.Rpc.name, call.Rpc.params with
+  | "Cluster_host.forget", [_session; self] ->
+    let open API in
+    Xapi_cluster_host.forget ~__context ~self:(ref_Cluster_host_of_rpc self);
+    Rpc.{success = true; contents = Rpc.String "" }
+  | "host.apply_guest_agent_config", _ ->
+    Rpc.{success = true; contents = Rpc.rpc_of_unit () }
+  | name, params ->
+    failwith (Printf.sprintf "Unexpected RPC: %s(%s)" name (String.concat " " (List.map Rpc.to_string params)))
+
+let make ~__context extra_hosts =
+  Context.set_test_rpc __context (test_rpc ~__context);
+  Context.set_test_clusterd_rpc __context (test_clusterd_rpc ~__context);
+  Test_common.make_cluster_and_hosts ~__context extra_hosts
+
+
+let test_forget () =
+  let __context = Test_common.make_test_database () in
+  let host2 = Test_common.make_host ~__context () in
+  let cluster, original_cluster_hosts = make ~__context [host2] in
+
+  Xapi_host.destroy ~__context ~self:host2;
+  let pending = Db.Cluster.get_pending_forget ~__context ~self:cluster in
+  Alcotest.(check (list string) "no pending forgets"
```

Comment:
You don't need to include the whole test in parentheses, just `Alcotest.(check (list string))`, although it's better to use `Alcotest.(check (slist string [String.compare or compare]))`.